pkg/system/mknod_freebsd.go

```diff
@@ -0,0 +1,20 @@
+//go:build freebsd
+// +build freebsd
+
+package system // import "github.com/docker/docker/pkg/system"
+
+import (
+	"golang.org/x/sys/unix"
+)
+
+// Mknod creates a filesystem node (file, device special file or named pipe) named path
+// with attributes specified by mode and dev.
+func Mknod(path string, mode uint32, dev int) error {
+	return unix.Mknod(path, mode, uint64(dev))
+}
+
+// Mkdev is used to build the value of FreeBSD devices (in /dev/) which specifies major
+// and minor number of the newly created device special file.
+func Mkdev(major int64, minor int64) uint64 {
+	return unix.Mkdev(uint32(major), uint32(minor))
```

Comment:
Doing a quick diff between both, to see what the difference is. On Linux we're converting this to return an `uint32`. All places where it's used (currently) cast it to an `int` (as argument for `Mknod`).

Mostly wondering if we should do the same for freebsd so that the signature is the same. (Perhaps we should look at that in a follow-up, and move these utilities into pkg/archive).

@samuelkarp perhaps you have thoughts on this?


```patch
diff --git a/pkg/system/mknod.go b/pkg/system/mknod_freebsd.go
index ccb2e638bb..b1ba8320dd 100644
--- a/pkg/system/mknod.go
+++ b/pkg/system/mknod_freebsd.go
@@ -1,5 +1,5 @@
-//go:build !freebsd && !windows
-// +build !freebsd,!windows
+//go:build freebsd
+// +build freebsd

 package system // import "github.com/docker/docker/pkg/system"

@@ -10,14 +10,11 @@ import (
 // Mknod creates a filesystem node (file, device special file or named pipe) named path
 // with attributes specified by mode and dev.
 func Mknod(path string, mode uint32, dev int) error {
-       return unix.Mknod(path, mode, dev)
+       return unix.Mknod(path, mode, uint64(dev))
 }

-// Mkdev is used to build the value of linux devices (in /dev/) which specifies major
+// Mkdev is used to build the value of FreeBSD devices (in /dev/) which specifies major
 // and minor number of the newly created device special file.
-// Linux device nodes are a bit weird due to backwards compat with 16 bit device nodes.
-// They are, from low to high: the lower 8 bits of the minor, then 12 bits of the major,
-// then the top 12 bits of the minor.
-func Mkdev(major int64, minor int64) uint32 {
-       return uint32(unix.Mkdev(uint32(major), uint32(minor)))
+func Mkdev(major int64, minor int64) uint64 {
+       return unix.Mkdev(uint32(major), uint32(minor))
 }
```

Reply:
Makes sense, thanks